### PR TITLE
disttae: add transaction workspace and check primary key

### PR DESCRIPTION
**Pull request by reusee**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #5950 

## What this PR does / why we need it:

### CODE REVIEW

**Review by nnsgmsone — APPROVED** [review]

File: pkg/vm/engine/disttae/txn.go [hunk 1]
```diff
@@ -212,8 +215,16 @@ func (txn *Transaction) IncStatementId() {
 
 // Write used to write data to the transaction buffer
 // insert/delete/update all use this api
-func (txn *Transaction) WriteBatch(typ int, databaseId, tableId uint64,
-	databaseName, tableName string, bat *batch.Batch, dnStore DNStore) error {
+func (txn *Transaction) WriteBatch(
```
Comment (fengttt) [inline pkg/vm/engine/disttae/txn.go:218]: this function, added param primaryIdx, and -1 all need to be documented.

**Review by fengttt — COMMENTED** [review]
Also need a better commit message, on what exactly this PR activated to catch the dup key earlier than before.

File: pkg/vm/engine/disttae/txn.go (hunk 1 above)
Comment (reusee) [reply pkg/vm/engine/disttae/txn.go:218]: added

**Review by fengttt — APPROVED** [review]

**Comment by fengttt** [comment]
@mergifyio refresh

**Comment by reusee** [comment]
I believe this PR fixes #5950 but breaks some cases in BVT. Those cases are reporting duplication on commit but this PR will make it report earlier. I will talk to Sudong to find out how to update BVT cases.

**Review by aressu1985 — APPROVED** [review]

**Pull request merged**